eng/pipelines/templates/jobs/sdk-job-matrix.yml

```diff
@@ -29,39 +29,20 @@ parameters:
     # Don't run the tests on arm64. Only perform the build itself.
     runTests: false
   - categoryName: ContainerBased
-    container: ubuntu2204DebPkg
-    helixTargetContainer: $(helixTargetContainerPrefix)ubuntu-22.04-helix-amd64
+    container: azureLinux30Net10BuildAmd64
+    helixTargetContainer: $(helixTargetContainerPrefix)azurelinux.3.amd64.open
     osProperties: $(linuxOsPortableProperties)
     runTests: true
   - categoryName: ContainerBased
-    container: fedora39
-    # No fedora Helix container is available, so use the ubuntu one instead.
-    helixTargetContainer: $(helixTargetContainerPrefix)ubuntu-22.04-helix-amd64
-    osProperties: $(linuxOsPortableProperties)
-    # Skipping all container-based testing for now.
-    # See: https://github.com/dotnet/sdk/issues/40935
-    runTests: false
-  - categoryName: ContainerBased
-    container: centosStream9
-    helixTargetContainer: $(helixTargetContainerPrefix)centos-stream9-helix
+    container: azureLinux30Net10BuildAmd64
+    helixTargetContainer: $(helixTargetContainerPrefix)azurelinux.3.amd64.open
     osProperties: /p:OSName=linux
     runTests: true
   - categoryName: ContainerBased
-    container: debian12Amd64
-    helixTargetContainer: $(helixTargetContainerPrefix)debian-11-helix-amd64
+    container: azureLinux30Net10BuildAmd64
+    helixTargetContainer: $(helixTargetContainerPrefix)azurelinux.3.amd64.open
     osProperties: /p:OSName=linux /p:BuildSdkDeb=true
```

Comment:
The `BuildSdkDeb` property can be changed to always be true when `OSName` is `linux`. There's no need to have a separate job for this now that we can build debs on any linux platform.

Reply:
@jkoritzinsky I removed the one obvious duplicate. There was another one that uses linuxOsglibcProperties which appear to just be setting OSName and IsLinuxglibc=true which is just used in our badging on the download table. Seems like I might be able to combine those as well?  I would have expected linuxOsglibcProperties to do more.

Reply:
I expect that a lot of the logic there was around installer building and validation that has since been removed.

You can collapse those together as well.